core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -812,7 +971,7 @@ class DAGScheduler(
         case s: ResultStage =>
           val job = s.resultOfJob.get
           partitionsToCompute.map { id =>
-            val p = job.partitions(id)
+            val p = s.partitions(id)
```

Comment:
I've wondered about this line many times in the past, but been too scared to change it -- is there any reason why `job.partitions` will be different from `resultStage.partitions`?